pkg/utils/utils.go

```diff
@@ -16,6 +16,23 @@ limitations under the License.
 
 package utils
 
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/names"
+)
+
+func CreateDeploymentOwnerRef(kubeClient *kubernetes.Clientset, managerNamespace string) ([]metav1.OwnerReference, error) {
```

Comment:
If you move `createOwnerRefForService` under `webhook`, this could be moved as well. This is weirdly specific function to live in `utils`.